### PR TITLE
PYI-353:  Make auth a rest api

### PR DESCRIPTION
**Pull request by dannyhayes-gds**
### What changed
Changed authorisation lambda to return auth code as a JSON payload in 200 request.

Refactored the Aws gateway response methods to handle the JSON conversion.

### Why did it change
Have the lambda act as a REST api response rather than the 302 redirect response.

### Issue tracking

- [PYI-353](https://govukverify.atlassian.net/browse/PYI-353)

### CODE REVIEW

File: src/test/java/uk/gov/di/ipv/lambda/AuthorizationHandlerTest.java [hunk 1]
```diff
@@ -56,11 +55,16 @@ public void shouldReturnAuthResponseOnSuccessfulOauthRequest() throws Exception
         params.put("client_id", "12345");
         params.put("response_type", "code");
         params.put("scope", "openid");
+        params.put("scope", "openid");
```
Comment (merlinc) [inline src/test/java/uk/gov/di/ipv/lambda/AuthorizationHandlerTest.java]: This line is duplicated :)

**Review by merlinc — COMMENTED** [review]
There's a duplicated line in the test, but the rest of it looks ok to me

File: src/test/java/uk/gov/di/ipv/lambda/AuthorizationHandlerTest.java (hunk 1 above)
Comment (dannyhayes-gds) [reply src/test/java/uk/gov/di/ipv/lambda/AuthorizationHandlerTest.java]: Ah yeah good spot. I'll remove this. Cheers

**Review by idrop — APPROVED** [review]
🚀

**Review by kerrr — APPROVED** [review]

**Pull request merged**